src/plugins/core/header_size.ts

```diff
@@ -15,7 +15,10 @@ export class HeaderSizePlugin extends CorePlugin<HeaderSizeState> implements Hea
   handle(cmd: Command) {
     switch (cmd.type) {
       case "CREATE_SHEET": {
-        this.history.update("sizes", cmd.sheetId, { COL: [], ROW: [] });
+        this.history.update("sizes", cmd.sheetId, {
```

Comment:
the plugin `HeaderGroupingPlugin` has the stame structure. It just so happens that it uses array.map  where this plugin uses array.filter. I guess it's worth applying the same fix preemptively

Reply:
The cases differ because in HeaderSizePlugin, sizes are stored at specific indexes in ROW and COL arrays, making initialization crucial to avoid sparse array issues.

In HeaderGroupingPlugin, however, we store groups with start and end ranges rather than at specific indexes, so the array's structure isn’t impacted in the same way since we don't store values directly at particular indexes.

I just checked, and applying the same fix broke several tests.

Reply:
Thanks for your analysis and explanation :)